cli_tools/gce_onestep_image_import/onestep_importer/aws_importer/importer.go

```diff
@@ -0,0 +1,350 @@
+	if err != nil {
+		return err
+	}
+
+	// 2. get export task id from response
+	var resp map[string]interface{}
+	if err := json.Unmarshal(output, &resp); err != nil {
+		return err
+	}
+	tid := resp["ExportImageTaskId"]
+	if tid == "" {
+		return fmt.Errorf("empty task id returned")
+	}
+
+	// 3. monitor export task progress
+	for {
+		// aws ec2 describe-export-image-tasks --export-image-task-ids <task id>
+		output, err = onestepUtils.RunCmdAndGetOutputWithoutLog("aws", []string{"ec2", "describe-export-image-tasks", "--export-image-task-ids", fmt.Sprintf("%v", tid)})
+		if err != nil {
+			return err
+		}
+		var taskResp awsTaskResponse
+		if err := json.Unmarshal(output, &taskResp); err != nil {
+			return err
+		}
+		if len(taskResp.ExportImageTasks) != 1 {
+			return fmt.Errorf("unexpected response of describe-export-image-tasks")
+		}
+		log.Println(fmt.Sprintf("AWS export task status: %v, status message: %v, progress: %v", taskResp.ExportImageTasks[0].Status, taskResp.ExportImageTasks[0].StatusMessage, taskResp.ExportImageTasks[0].Progress))
+
+		if taskResp.ExportImageTasks[0].Status != "active" {
+			if taskResp.ExportImageTasks[0].Status != "completed" {
+				return fmt.Errorf("AWS export task wasn't completed successfully")
+			}
+			log.Println("AWS export task is completed!")
+			break
+		}
+		time.Sleep(time.Millisecond * 3000)
+	}
+
+	// 4. set exported file data
+	importer.args.ExportKey = fmt.Sprintf("%v%v.vmdk", importer.args.ExportFolder, tid)
+	importer.args.ExportedAMIPath = fmt.Sprintf("s3://%v/%v", importer.args.ExportBucket, importer.args.ExportKey)
+	log.Println(fmt.Sprintf("Exported location is %v.", importer.args.ExportedAMIPath))
+
+	return nil
+}
+
+// copyToGCS copies vmdk file from S3 to GCS
+func (importer *AWSImporter) copyToGCS() (string, error) {
+	// 1. get GCS path as copy destination.
+	gcsFilePath := pathutils.JoinURL(importer.args.GcsScratchBucket,
+		fmt.Sprintf("onestep-image-import-aws-%v.vmdk", pathutils.RandString(5)))
+
+	log.Println(fmt.Sprintf("Copying %v to %v.",
+		importer.args.ExportedAMIPath, gcsFilePath))
+
+	// 2. create a new folder for local buffer
+	path := filepath.Join(filepath.Dir(importer.args.ExecutablePath), fmt.Sprint("upload", pathutils.RandString(5)))
+
+	err := os.Mkdir(path, 0755)
+	if err != nil {
+		return "", err
+	}
+	defer os.RemoveAll(path)
+
+	// 3. get writer
+	bs, err := humanize.ParseBytes(uploadBufSize)
+	if err != nil {
+		return "", err
+	}
+	bkt, obj, err := storageutils.GetGCSObjectPathElements(gcsFilePath)
+	if err != nil {
+		return "", err
+	}
+	workers := int64(runtime.NumCPU())
+	writer := storageutils.NewBufferedWriter(importer.ctx, int64(bs), workers, gcsClient, importer.oauth, path, bkt, obj)
+
+	return gcsFilePath, importer.stream(writer)
+}
+
+// stream downloads S3 file and uploads to GCS concurrently
+func (importer *AWSImporter) stream(writer io.WriteCloser) error {
```

Comment:
"stream" is too general, find a name that is more descriptive

Reply:
I renamed to transferFile(). What do you think about this name?